crates/bevy_pbr/src/render/pbr_lighting.wgsl

```diff
@@ -284,3 +285,155 @@ fn directional_light(light_id: u32, roughness: f32, NdotV: f32, normal: vec3<f32
+    // roughness) or inside the material (due to subsurface scattering). Here, we only consider
+    // the first scenario.
+    //
+    // Blur intensity is:
+    // - squarely proportional to `perceptual_roughness`
+    // - inversely proportional to view z
+    let blur_intensity = (perceptual_roughness * perceptual_roughness) / view_z;
+
+#ifdef TRANSMISSIVE_TAPS
+    let num_taps = #{TRANSMISSIVE_TAPS}; // Controlled by the `Camera3d::transmissive_quality` property
+#else
+    let num_taps = 8; // Fallback to 8 taps
+#endif
+    let num_spirals = i32(ceil(f32(num_taps) / 8.0));
+    let random_angle = interleaved_gradient_noise(frag_coord.xy);
+
+    // Pixel checkerboard pattern (helps make the interleaved gradient noise pattern less visible)
+    let pixel_checkboard = (
+#ifdef TEMPORAL_JITTER
+        // 0 or 1 on even/odd pixels, alternates every frame
+        (i32(frag_coord.x) + i32(frag_coord.y) + i32(view_bindings::globals.frame_count)) % 2
+#else
+        // 0 or 1 on even/odd pixels
+        (i32(frag_coord.x) + i32(frag_coord.y)) % 2
+#endif
+    );
+
+    var result = vec4<f32>(0.0);
+    for (var i: i32 = 0; i < num_taps; i = i + 1) {
+        let current_spiral = (i >> 3u);
+        let angle = (random_angle + f32(current_spiral) / f32(num_spirals)) * 2.0 * PI;
+        let m = vec2(sin(angle), cos(angle));
+        let rotation_matrix = mat2x2(
+            m.y, -m.x,
+            m.x, m.y
+        );
+
+        // Get spiral offset
+        var spiral_offset: vec2<f32>;
+        switch i & 7 {
+            // https://www.iryoku.com/next-generation-post-processing-in-call-of-duty-advanced-warfare (slides 120-135)
+            // TODO: Figure out a more reasonable way of doing this, as WGSL
+            // seems to only allow constant indexes into constant arrays
+            case 0: { spiral_offset = vec2<f32>(-0.7071,  0.7071); }
+            case 1: { spiral_offset = vec2<f32>(-0.0000, -0.8750); }
+            case 2: { spiral_offset = vec2<f32>( 0.5303,  0.5303); }
+            case 3: { spiral_offset = vec2<f32>(-0.6250, -0.0000); }
+            case 4: { spiral_offset = vec2<f32>( 0.3536, -0.3536); }
+            case 5: { spiral_offset = vec2<f32>(-0.0000,  0.3750); }
+            case 6: { spiral_offset = vec2<f32>(-0.1768, -0.1768); }
+            case 7: { spiral_offset = vec2<f32>( 0.1250,  0.0000); }
+            default: {}
+        }
+
+        // Make each consecutive spiral slightly smaller than the previous one
+        spiral_offset *= 1.0 - (0.5 * f32(current_spiral + 1) / f32(num_spirals));
+
+        // Rotate and correct for aspect ratio
+        let rotated_spiral_offset = (rotation_matrix * spiral_offset) * vec2(1.0, aspect);
+
+        // Calculate final offset position, with blur and spiral offset
+        let modified_offset_position = offset_position + rotated_spiral_offset * blur_intensity * (1.0 - f32(pixel_checkboard) * 0.1);
+
+#ifdef PREPASS_DEPTH_SUPPORTED
```

Comment:
This doesn't seem right. You need to check if the prepass depth is actually enabled... `DEPTH_PREPASS`.

Reply:
AFAIK there's currently no shader def to check whether the depth prepass is enabled from the main pass.

There's:

- `DEPTH_PREPASS` which is only set if we're currently rendering the depth prepass;
- `DEPTH_PREPASS_SUPPORTED` (Added by this PR) which is set if the current configuration supports a depth prepass;

I'm not entirely sure how to best set up the plumbing to expose a `DEPTH_PREPASS_ENABLED` shaderdef to the shader